drivers/platform/stm32/stm32_spi.c

```diff
@@ -386,12 +424,134 @@ int32_t stm32_spi_write_and_read(struct no_os_spi_desc *desc,
 	return stm32_spi_transfer(desc, &msg, 1);
 }
 
+/**
+ * @brief Configure and start a series of transfers using DMA.
+ * @param desc - The SPI descriptor.
+ * @param msgs - The messages array.
+ * @param len - Number of messages.
+ * @return 0 in case of success, errno codes otherwise.
+ */
+int32_t stm32_config_dma_and_start(struct no_os_spi_desc* desc,
+				   struct no_os_spi_msg* msgs,
+				   uint32_t len)
+{
+	struct stm32_spi_desc* sdesc = desc->extra;
+	struct no_os_dma_xfer_desc* rx_ch_xfer;
+	struct no_os_dma_xfer_desc* tx_ch_xfer;
+	struct no_os_dma_ch* tx_ch = sdesc->txdma_ch;
+	struct no_os_dma_ch* rx_ch = sdesc->rxdma_ch;
+	struct stm32_dma_channel* sdma_rx = rx_ch->extra;
+	struct stm32_dma_channel* sdma_tx = tx_ch->extra;
+	SPI_TypeDef* SPIx = sdesc->hspi.Instance;
+	int ret;
+	uint8_t i;
+
+	if (!desc || !msgs)
+		return -EINVAL;
+
+	rx_ch_xfer = no_os_calloc(len, sizeof(*rx_ch_xfer));
+	if (!rx_ch_xfer)
+		return -ENOMEM;
+
+	tx_ch_xfer = no_os_calloc(len, sizeof(*tx_ch_xfer));
+	if (!tx_ch_xfer) {
+		goto free_rx_ch_xfer;
+	}
+
+	for (i = 0; i < len; i++) {
+		tx_ch_xfer[i].src = msgs[i].tx_buff;
+		tx_ch_xfer[i].dst = &(SPIx->DR);
+		tx_ch_xfer[i].xfer_type = MEM_TO_DEV;
+		tx_ch_xfer[i].periph = NO_OS_DMA_IRQ;
+		tx_ch_xfer[i].length = msgs[i].bytes_number;
+
+		rx_ch_xfer[i].dst = msgs[i].rx_buff;
+		rx_ch_xfer[i].src = &(SPIx->DR);
+		rx_ch_xfer[i].periph = NO_OS_DMA_IRQ;
+		rx_ch_xfer[i].xfer_type = DEV_TO_MEM;
+		rx_ch_xfer[i].length = msgs[i].bytes_number;
+	}
+
+	rx_ch->id = sdma_rx->hdma;
+	rx_ch->sg_list = NULL;
+	if (sdma_rx->irq_num > 0)
```

Comment:
The `irq_num` is `uint32_t`. Moreover, I think this adds unnecessary complexity. I don't see why would the user care to assign a specific interrupt number.

Reply:
irq_num is a field required in the no_os_irq_register_callback()/ no_os_irq_set_priority(). These are being called in the no_os_dma_config_xfer() for configuring transfer complete callbacks